Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,138 @@
+            self.exchangeRatesAPIKey = apiKeyFromPlistFile
+        }
+
+        if let urlSession = urlSession {
+            self.networkManager = ExchangeRatesNetworkManager(session: urlSession)
+        } else {
+            self.networkManager = ExchangeRatesNetworkManager(session: URLSession.shared)
+        }
     }
+    
+    public func getExchangeRate(
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exhangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    completion(.success(rates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrency: Currency,
+        completion: @escaping (Result<Double, Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestConvert(
+                    apiKey: exchangeRatesAPIKey, url: convertURL.get(), amount: amount, base: baseCurrency, target: targetCurrency)
+                switch result {
+                case .success(let result):
+                    completion(.success(result))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exhangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    var modifiedRates = rates
+                    for rate in modifiedRates {
+                        let value = rate.value * amount
+                        modifiedRates.updateValue(value, forKey: rate.key)
+                    }
+                    completion(.success(modifiedRates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
 }
+
+private extension SimpleCurrencyConverter {
+    
+    var apiKeyFromPlistFile: String {
+        guard let key = Bundle.main.infoDictionary?["ExchangeRatesAPIKey"] as? String else {
+            fatalError("❗ Failed to get exchangerates.io API key from info.plist file ❗")
```

Comment:
`assertionFailure()` is more preferable here (see my other comment in `setup()` method)